src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs

```diff
@@ -453,8 +453,12 @@ internal static class DependentTypeFinder
             // implement that type.  Because the mapping is from the simple name
             // we might get false positives.  But that's fine as we still use 
             // 'metadataTypeMatches' to make sure the match is correct.
-            var symbolTreeInfo = await SymbolTreeInfo.GetInfoForMetadataReferenceAsync(
+            var symbolTreeInfo = await SymbolTreeInfo.TryGetInfoForMetadataReferenceAsync(
                 project.Solution, reference, loadOnly: false, cancellationToken: cancellationToken).ConfigureAwait(false);
+            if (symbolTreeInfo == null)
+            {
+                return;
+            }
```

Comment:
Not simply.  But if you have a bogus metadata dll, we could attempt to check it in, then run this scenario.